src/core/hdf_simulation_store.jl

```diff
@@ -458,19 +473,23 @@ function _get_columns_dataset(store::HdfSimulationStore, key)
     return getfield(store.datasets[key.stage], key.type)[key.name].columns
 end
 
-function _get_dataset(::Type{OptimizerStats}, store::HdfSimulationStore)
+function get_dataset(::Type{OptimizerStats}, store::HdfSimulationStore)
     return store.file[OPTIMIZER_DATASET_PATH]
 end
 
-function _get_dataset(store::HdfSimulationStore, key)
+function get_dataset(store::HdfSimulationStore, stage_name::Symbol)
```

Comment:
@daniel-thom I am not sure if you might approve of this. I welcome suggestions about how to get the available symbols in a dataset.

Reply:
I don't think this should be called outside of this file. We can leave it for now and I'll refactor next week when I add the `list_names` and other interfaces.

Reply:
ok, yeah I wasn't sure the best approach here to get this but now it has been clarified.